framework/Samplers/Umbrella.py

```diff
@@ -0,0 +1,311 @@
+import numpy as np
+from operator import mul
+from functools import reduce, partial
+import scipy
+import pandas as pd
+from scipy.stats import gamma, norm, multivariate_normal
+import operator
+import math
+#External Modules End--------------------------------------------------------------------------------
+
+#Internal Modules------------------------------------------------------------------------------------
+from .ForwardSampler import ForwardSampler
+from utils import utils,randomUtils,InputData, InputTypes
+#Internal Modules End--------------------------------------------------------------------------------
+
+class Umbrella(ForwardSampler):
+  """
+    MONTE CARLO Sampler
+  """
+
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for
+      class cls.
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for
+        specifying input of cls.
+    """
+    inputSpecification = super(Umbrella, cls).getInputSpecification()
+    umbrellaInput = InputData.parameterInputFactory("umbrella", contentType=InputTypes.StringType)
+    samplerInitInput = InputData.parameterInputFactory("samplerInit")
+    dimInput = InputData.parameterInputFactory("dimensions", contentType=InputTypes.IntegerType)
+    samplerInitInput.addSub(dimInput)
+    limit = InputData.parameterInputFactory("limit", contentType=InputTypes.IntegerType)
+    samplerInitInput.addSub(limit)
+    modeLocationInput = InputData.parameterInputFactory("modeLocation", contentType=InputTypes.IntegerType)
+    samplerInitInput.addSub(modeLocationInput)
+    tailProbabilityInput = InputData.parameterInputFactory("tailProbability", contentType=InputTypes.FloatType)
+    samplerInitInput.addSub(tailProbabilityInput)
+    weightTargetInput = InputData.parameterInputFactory("weightTarget", contentType=InputTypes.FloatType)
+    samplerInitInput.addSub(weightTargetInput)
+    rhoInput = InputData.parameterInputFactory("rho", contentType=InputTypes.FloatType)
+    samplerInitInput.addSub(rhoInput)
+    targetDistribution = InputData.parameterInputFactory("distribution")
+    samplerInitInput.addSub(targetDistribution)
+    samplingTypeInput = InputData.parameterInputFactory("samplingType", contentType=InputTypes.StringType)
+    samplerInitInput.addSub(samplingTypeInput)
+    inputSpecification.addSub(samplerInitInput)
+
+    return inputSpecification
+
+  def __init__(self):
+    """
+      Default Constructor that will initialize member variables with reasonable
+      defaults or empty lists/dictionaries where applicable.
+      @ In, None
+      @ Out, None
+    """
+    ForwardSampler.__init__(self)
+    self.printTag = 'SAMPLER UMBRELLA'
+    self.samplingType = None
+    self.limit = None
+    self.modeLocation = 0
+    self.rho = 0
+    self.dimension = 0
```

Comment:
Variables introduced in the constructor should have a comment after them describing the variable; for example, "rho" is not clear unless the dev is familiar with the literature for the method